### PR TITLE
Add `copy` for `DataGroup`

### LINKED ISSUE #2964: [DataGroup] Add `copy` method
Should also support the `deep: bool` option just like for other scipp objects, I think?

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes #2964.

### CODE REVIEW

File: src/scipp/core/data_group.py [hunk 1]
```diff
@@ -57,6 +58,9 @@ def __init__(self, /, *args, **kwargs):
         if not all([isinstance(k, str) for k in self._items.keys()]):
             raise ValueError("DataGroup keys must be strings.")
 
+    def __copy__(self) -> DataGroup:
+        return DataGroup(copy.copy(self._items))
+
```
Comment (jl-wynen) [inline src/scipp/core/data_group.py:63]: What about `__deepcopy__`? Does it do the right thing by default?

File: src/scipp/core/data_group.py [hunk 2]
```diff
@@ -191,49 +195,48 @@ def __add__(self, other):
...
-        return DataGroup({key: func(value) for key, value in self.items()})
-
     @property
     def bins(self):
         # TODO Returning a regular DataGroup here may be wrong, since the `bins`
         # property provides a different set of attrs and methods.
-        return self._call_method(operator.attrgetter('bins'))
+        return self.apply(operator.attrgetter('bins'))
+
+    def apply(self, func: Callable, *args, **kwargs) -> DataGroup:
```
Comment (jl-wynen) [inline src/scipp/core/data_group.py:204]: Are the extra arguments necessary? We could require callers to use `partial` or similar.

File: src/scipp/core/data_group.py (hunk 1 above)
Comment (SimonHeybrock) [reply src/scipp/core/data_group.py:63]: Yes, see tests.

File: src/scipp/core/data_group.py (hunk 2 above)
Comment (SimonHeybrock) [reply src/scipp/core/data_group.py:204]: Are they a problem? I used a signature similar to Pandas.

Comment (jl-wynen) [reply src/scipp/core/data_group.py:204]: Not a problem. But it would simplify the interface. Leave it, I just wanted to know your motivation.

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**